### PR TITLE
Add DeepGEMM prerelease wheel tests

### PR DESCRIPTION
**Pull request by b8zhong**
## Motivation

To avoid cases like https://github.com/sgl-project/sglang/issues/26541, and https://github.com/deepseek-ai/DeepGEMM/pull/342

Here is what is completed vs. what needs to be done (since PDL functional correctness needs to be E2E, we ignore it):
So we still need to add test for FP8 combine, and DG_USE_MXF4_KIND (in SGL DeepGEMM repo). Table is Claude, should be accurate

  ## sgl-deep-gemm fork additions

  | Fork addition | PR/commit | Arch | Covered | Test(s) |
  |---|---|---|:---:|---|
  | tvm-ffi binding (`tvm_ffi_api.cpp`, `_C`) | `ffb3d29` | SM90 + SM100 | ✅ | all tests call `deep_gemm.*` through `_C` |
  | BF16 grouped GEMM wrappers | `48eb6a6` | SM90 + SM100 | ✅ | `test_bf16` (m_grouped contiguous/masked, `use_bf16`) |
  | FP4 acts + fused `mega_moe_pre_dispatch` | #27 | SM100 | ✅ | `test_mega_moe_pre_dispatch` (`dtype=both`), `test_mega_moe_l1_sentinel`, `test_mega_moe_l1_fp4_accuracy` |
  | w4a4 megamoe tvm-ffi wrapper | `8ab0096` | SM100 | ✅ | `test_mega_moe_l1_*` |
  | paged MQA logits IMA guard | #38 | SM90 + SM100 | ✅ | `test_attention` (paged mqa logits) |
  | PDL utils export (`set_pdl`/`get_pdl`) | #34 | SM90 + SM100 | ⚪ | out of scope |
  | **`DG_USE_FP8_COMBINE` (FP8 combine path)** | #28 | SM100 | ❌ | **none** — env-gated, needs new test in DeepGEMM repo |
  | **`DG_USE_MXF4_KIND` (MXF4 mainloop)** | #27 | SM100 | ❌ | **none** — env-gated, needs new test in DeepGEMM repo |

  **Legend:** ✅ covered · ❌ gap · ⚪ intentionally out of scope

---
### CI States

Latest PR Test (Base): :white_check_mark: [Run #27358564170](https://github.com/sgl-project/sglang/actions/runs/27358564170)
Latest PR Test (Extra): :x: [Run #27358567647](https://github.com/sgl-project/sglang/actions/runs/27358567647)

### CODE REVIEW

File: .github/workflows/release-whl-deepgemm.yml [hunk 1]
```diff
@@ -74,8 +74,53 @@ jobs:
...
+    runs-on: ${{ matrix.runner }}
+    timeout-minutes: 120
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Checkout DeepGEMM
+        uses: actions/checkout@v4
+        with:
+          repository: sgl-project/DeepGEMM
+          ref: ${{ inputs.branch }}
```
Comment (Fridge003) [inline .github/workflows/release-whl-deepgemm.yml]: The branch here for testing can be fixed to `dev`

File: .github/workflows/release-whl-deepgemm.yml [hunk 2]
```diff
@@ -159,20 +204,65 @@ jobs:
...
+        with:
+          path: dist/
+          merge-multiple: true
+          pattern: deepgemm-wheel-cuda13.0-x86_64
+
+      - name: Install wheel and deps
+        run: |
+          python3 -m pip install --upgrade pip
+          # torch must match the version/ABI the bundled _C.so was built against.
+          python3 -m pip install torch==2.11.0 --index-url https://download.pytorch.org/whl/cu130
```
Comment (Fridge003) [inline .github/workflows/release-whl-deepgemm.yml]: Can we manage these dependencies on deepgemm side?

File: .github/workflows/release-whl-deepgemm.yml [hunk 3]
```diff
@@ -74,8 +74,53 @@ jobs:
...
-  release-cu129:
+  test-cu129:
     needs: build-cu129-matrix
+    # Only the x86_64 wheel is GPU-validated; the aarch64 wheel ships unvalidated.
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch_label: sm90
+            runner: 8-gpu-h200
```
Comment (Fridge003) [inline .github/workflows/release-whl-deepgemm.yml:89]: Can we reduce the number of gpus for testing to 4 or 2

**Comment by Fridge003** [comment]
For testing fp4 megamoe: https://github.com/sgl-project/DeepGEMM/blob/dev/tests/test_mega_moe_l1_fp4_accuracy.py

File: .github/workflows/release-whl-deepgemm.yml (hunk 1 above)
Comment (b8zhong) [reply .github/workflows/release-whl-deepgemm.yml]: Yes

File: .github/workflows/release-whl-deepgemm.yml (hunk 2 above)
Comment (b8zhong) [reply .github/workflows/release-whl-deepgemm.yml]: Technically, yeah. But if the runner torch and the build torch is unaligned, then it will cause problems (like undefined symbols). So just add an assertion to be clear

Comment (Fridge003) [reply .github/workflows/release-whl-deepgemm.yml]: OK

**Comment by b8zhong** [comment]
Looks like we do need to re-install Torch in case of mismatch:

https://github.com/sgl-project/sglang/actions/runs/27299804222/job/80643354800
Screenshot 2026-06-10 at 12 54 14 PM

Thus, new run tracked here: https://github.com/sgl-project/sglang/actions/runs/27302299942

File: .github/workflows/release-whl-deepgemm.yml [hunk 4]
```diff
@@ -74,8 +79,56 @@ jobs:
...
 
-  release-cu129:
+  test-cu129:
     needs: build-cu129-matrix
+    # Only the x86_64 wheel is GPU-validated; the aarch64 wheel ships unvalidated.
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch_label: sm90
```
Comment (Fridge003) [inline .github/workflows/release-whl-deepgemm.yml:88]: Maybe add sm103 test with 4-gpu-gb300 runner

**Comment by b8zhong** [comment]
With GB300: https://github.com/sgl-project/sglang/actions/runs/27318947491

**Comment by b8zhong** [comment]
https://github.com/sgl-project/sglang/actions/runs/27358576346

**Review by Fridge003 — APPROVED** [review]
